### ISSUE TITLE
bug: Wagmi 0.7.8 isn't working with react 17

### ISSUE THREAD

**Issue by Deveshb15** [wagmi-dev/wagmi#1199]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

0.7.8

### Current Behavior

I had wagmi 0.5.12 before but didn't realise that it didn't have useContract hook so I updated my wagmi package and I get this error in the log for some reason and the app doesn't work
![image](https://user-images.githubusercontent.com/72264218/198880409-f1cca1ef-f982-4b9c-9c55-283beabe373b.png)

### Expected Behavior

Basically it should've worked without any issues, I followed the docs too for createClient and using useConnect hook but it still doesn't work with react 17

**Comment by jxom** [comment]
Thanks for reporting! Can you please provide a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) via a CodeSandbox (feel free to use [this template](https://codesandbox.io/s/bitter-wind-pv0hwj)). I'll reopen this issue when a reproduction is provided! Sorry for the inconvenience.
